spec/features/daterange_picker_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+RSpec.shared_examples 'a model with date range validations' do
+  context 'validations' do
+    it 'checks whether required attribute values are present' do
```

Comment:
Maybe just name it would make it easier to understand
```suggestion
    it 'does not allow year_from to be blank' do
```